Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityProfileCloneWindow.cs

```diff
@@ -0,0 +1,320 @@
+﻿using Microsoft.MixedReality.Toolkit.Core.Definitions;
+using Microsoft.MixedReality.Toolkit.Core.Extensions.EditorClassExtensions;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Inspectors.Profiles
+{
+    public class MixedRealityProfileCloneWindow : EditorWindow
+    {
+        public enum ProfileCloneBehavior
+        {
+            UseExisting,        // Use the existing reference
+            CloneExisting,      // Create a clone of the sub-profile
+            UseSubstitution,    // Manually select a profile
+            LeaveEmpty,         // Set the reference to null
+        }
+
+        private struct SubProfileAction
+        {
+            public SubProfileAction(ProfileCloneBehavior behavior, SerializedProperty property, Object substitutionReference, System.Type profileType)
+            {
+                Behavior = behavior;
+                Property = property;
+                SubstitutionReference = substitutionReference;
+                ProfileType = profileType;
+
+                CloneName = (SubstitutionReference != null) ? "New " + SubstitutionReference.name : "New " + profileType.Name;
+            }
+
+            public ProfileCloneBehavior Behavior;
+            public SerializedProperty Property;
+            public string CloneName;
+            public Object SubstitutionReference;
+            public System.Type ProfileType;
+        }
+
+        private const string IsCustomProfileProperty = "isCustomProfile";
+        private static readonly Vector2 MinWindowSizeBasic = new Vector2(500, 140);
+        private const float SubProfileSizeMultiplier = 70f;
+        private static MixedRealityProfileCloneWindow cloneWindow;
+
+        private BaseMixedRealityProfile parentProfile;
+        private BaseMixedRealityProfile childProfile;
+        private SerializedProperty childProperty;
+        private SerializedObject childSerializedObject;
+        private Object targetFolder;
+        private string childProfileTypeName;
+        private string childProfileAssetName;
+        private List<SubProfileAction> subProfileActions = new List<SubProfileAction>();
+
+        public static void OpenWindow(BaseMixedRealityProfile parentProfile, BaseMixedRealityProfile childProfile, SerializedProperty childProperty)
+        {
+            if (cloneWindow != null)
+            {
+                cloneWindow.Close();
```

Comment:
Compared to closing this, is there a way to make the existing window receive foreground? If someone has unsaved changes with the window, I imagine theyll currently be lost by this ux choice to close the window?

Reply:
granted i guess if this call is made populating profiles, this may not be possible. Feel free to share any personal preference on which is more annoying: losing an already open window vs resetting focus to an open window with the wrong profiles :)

Reply:
I could launch a Proceed / Cancel dialog to verify that they want to stop cloning the current profile and instead clone another.